runtime/near-vm-logic/src/tests/context.rs

```diff
@@ -1,103 +1,89 @@
 use crate::tests::vm_logic_builder::VMLogicBuilder;
-use crate::VMContext;
-
-pub fn create_context() -> VMContext {
-    VMContext {
-        current_account_id: "alice".parse().unwrap(),
-        signer_account_id: "bob".parse().unwrap(),
-        signer_account_pk: vec![0, 1, 2, 3, 4],
-        predecessor_account_id: "carol".parse().unwrap(),
-        input: vec![0, 1, 2, 3, 5],
-        block_height: 10,
-        block_timestamp: 42,
-        epoch_height: 1,
-        account_balance: 2u128,
-        account_locked_balance: 1u128,
-        storage_usage: 12,
-        attached_deposit: 2u128,
-        prepaid_gas: 10_u64.pow(14),
-        random_seed: vec![0, 1, 2],
-        view_config: None,
-        output_data_receivers: vec![],
-    }
-}
 
 macro_rules! decl_test_bytes {
-    ($testname:ident, $method:ident, $input:expr) => {
+    ($testname:ident, $method:ident, $ctx:ident, $want:expr) => {
         #[test]
         fn $testname() {
             let mut logic_builder = VMLogicBuilder::default();
-            let mut logic = logic_builder.build(create_context());
+            let $ctx = &logic_builder.context;
+            let want = $want.to_vec();
+            let mut logic = logic_builder.build();
             logic.$method(0).expect("read bytes into register from context should be ok");
-            logic.assert_read_register($input, 0);
+            logic.assert_read_register(&want[..], 0);
```

Comment:
Thank you @nagisa 
Hm yeah, I agree it's probably fine to keep as is. I don't see the alternatives as a huge improvement.

Reply:
Alas those don’t work.  The former doesn’t because `ctx.<expression>` is not a valid Rust expression.  Trying that with `tt` doesn’t work because then macro only matches `current_account_id` only.  The latter doesn’t work because using `ctx` is unhygienic.

Reply:
I felt like testing this out, 

```diff
diff --git a/runtime/near-vm-logic/src/tests/context.rs b/runtime/near-vm-logic/src/tests/context.rs
index 445533a06..6af51017a 100644
--- a/runtime/near-vm-logic/src/tests/context.rs
+++ b/runtime/near-vm-logic/src/tests/context.rs
@@ -1,12 +1,12 @@
 use crate::tests::vm_logic_builder::VMLogicBuilder;
 
 macro_rules! decl_test_bytes {
-    ($testname:ident, $method:ident, $ctx:ident, $want:expr) => {
+    ($testname:ident, $method:ident, ctx . $($want:tt)*) => {
         #[test]
         fn $testname() {
             let mut logic_builder = VMLogicBuilder::default();
-            let $ctx = &logic_builder.context;
-            let want = $want.to_vec();
+            let ctx = &logic_builder.context;
+            let want = ctx.$($want)*.to_vec();
             let mut logic = logic_builder.build();
             logic.$method(0).expect("read bytes into register from context should be ok");
             logic.assert_read_register(&want[..], 0);
@@ -48,26 +48,23 @@ macro_rules! decl_test_u128 {
 decl_test_bytes!(
     test_current_account_id,
     current_account_id,
-    ctx,
     ctx.current_account_id.as_ref().as_bytes()
 );
 decl_test_bytes!(
     test_signer_account_id,
     signer_account_id,
-    ctx,
     ctx.signer_account_id.as_ref().as_bytes()
 );
 decl_test_bytes!(
     test_predecessor_account_id,
     predecessor_account_id,
-    ctx,
     ctx.predecessor_account_id.as_ref().as_bytes()
 );
-decl_test_bytes!(test_signer_account_pk, signer_account_pk, ctx, ctx.signer_account_pk);
+decl_test_bytes!(test_signer_account_pk, signer_account_pk, ctx.signer_account_pk);
 
-decl_test_bytes!(test_random_seed, random_seed, ctx, ctx.random_seed);
+decl_test_bytes!(test_random_seed, random_seed, ctx.random_seed);
 
-decl_test_bytes!(test_input, input, ctx, ctx.input);
+decl_test_bytes!(test_input, input, ctx.input);
```

builds and runs. But as I said, I don't believe this is something we want to do anyway. I think the current approach is clearer, less magic.